ygot/render_test.go

```diff
@@ -1873,6 +1874,67 @@ func TestConstructJSON(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "rewrite module name for an element with children",
+		in: &diffModAtRoot{
+			Child: &diffModAtRootChild{
+				ValueOne:   String("one"),
+				ValueTwo:   String("two"),
+				ValueThree: String("three"),
+			},
+			Elem: &diffModAtRootElem{
+				C: &diffModAtRootElemTwo{
+					Name: String("baz"),
+				},
+			},
+		},
+		inAppendMod: true,
+		inRewriteModuleNameRules: map[string]string{
+			// rewrite m1 to m2
+			"m1": "m2",
+		},
+		wantIETF: map[string]interface{}{
+			"m2:foo": map[string]interface{}{
+				"value-one":    "one",
+				"m3:value-two": "two",
+				"value-three":  "three",
```

Comment:
Should "value-three" be "m1:value-three"?

Reply:
No - since we rewrote m1->m2 then this becomes `m2:value-three`, but since we're in `m2` then we write this back to `m1`.

Reply:
Ahh I see, thanks for clarifying.

Reply:
No problem! Was there anything else you wanted to clarify here? Thanks for looking!

Reply:
Nope everything looks good, approved!